src/main.rs

```diff
@@ -132,7 +136,11 @@ fn build(app: &gtk::Application, opts: &Options) {
     nvim.ui_attach(80, 30, &ui_opts)
         .expect("Failed to attach UI");
 
-    let ui = ui::UI::init(app, rx, Rc::new(RefCell::new(nvim)));
+    let geometry = opts.geometry.split("x").collect::<Vec<&str>>();
+    let window_size = (geometry[0].parse::<i32>().unwrap(),
+                       geometry[1].parse::<i32>().unwrap());
```

Comment:
To handle the possible errors here with parsing to `i32`’s just do this instead:
```rust
    let window_size = (geometry[0].parse::<i32>().unwrap_or(1280),
	               geometry[1].parse::<i32>().unwrap_or(720));
```

`unwrap_or(1280)` is essentially saying, “Hey, try and get a value from this. If it’s an error and not a value, just give me 1280,” and likewise for `unwrap_or(720)`.

Reply:
That's fine, but what happens if I do `--geometry 1280` or `--geometry whatever`? I'm trying to figure out how to fail gracefully on line 139.

Reply:
Also, I'd rather error out with `Invalid geometry!" than silently revert to a default.

Reply:
Oh, right; sorry for misunderstanding the problem! What happens right now if you pass `--geometry whatever`?

Reply:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }', src/libcore/result.rs:1084:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
fish: “gnvim --geometry whateve” terminated by signal SIGILL (Illegal instruction)
```

Reply:
How about something like this:

```rust
    let window_size = (geometry[0].parse::<i32>().unwrap_or_else(panic!(“Invalid geometry!”), ...)
```

or something like:

```rust
geometry[1].parse::<i32>().unwrap_or_else(|| {
        println!(“Invalid geometry! Defaulting to 1280x720”);
        720 // or maybe `return;` if we just want to exit GNvim
}));
```

Reply:
That's certainly nicer, but doesn't cover the `split` error (which is the hardest to catch).

Reply:
Yeah, for sure. This is ugly, but I think it’d work:
```rust
if geometry.len() != 2 {
   println!(“Invalid geometry!”);
   return // Or whatever.
}
```

Or: 
```rust
 let window_size = if geometry.len() != 2 {
    (1280, 720)
} else {
    // Parse to i32’s and make into tuple and return it
}
```

Reply:
`suggestion
    let window_size = (geometry[0].parse::<i32>().unwrap_or(1280),
	               geometry[1].parse::<i32>().unwrap_or(720));
`

Reply:
No, `unwrap_or_else` doesn't seem to work like this. (And it complains about `return` in a function that has return types.)